### PR TITLE
feat: deploy on optimism kovan 

### LINKED ISSUE #154: Add Optimism testnet to factory deploy scripts
**User Story:**
As a Round Manager engineer
I want a script to deploy the RM factory contracts to Optimism Kovan
So that I can consistently deploy the factory contracts

**Acceptance Criteria**
- update deploy scripts to deploy factories on Optimism Kovan
- share link to factory contract on Optimism block explorer
- create documentation for updating deploy scripts

**Product & Design Links:**

**Tech Details:**
- Review [deploy steps here](https://github.com/gitcoinco/grants-round/blob/main/packages/contracts/docs/DEPLOY_STEPS.md)

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### LINKED ISSUE #155: Add Optimism testnet to implementation deploy scripts
**User Story:**
As a Round Manager engineer
I want a script to deploy the RM implementation contracts to Optimism Kovan
So that I can consistently deploy the implementation contracts

**Acceptance Criteria**
- update deploy scripts to deploy implementation contracts on Optimism Kovan
- share link to implementation contract on Optimism Kovan block explorer

**Product & Design Links:**

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### LINKED ISSUE #156: Create program and round on Optimism testnet via script
**Overview**
As a round manager engineer
I want to deploy a program and round via script
So that I know that the factory and implementation contracts work

**Acceptance criteria**
- Share links to program and round contracts on Optimism

**Details**
- Documentation available [here](https://github.com/gitcoinco/grants-round/blob/main/packages/contracts/docs/DEPLOY_STEPS.md)

### PR DESCRIPTION
**Pull request by thelostone-mc**
### Description

- update script command to support multiple networks
- add doc to state-supported networks
- update the documentation to reflect updated scripts
- add programContract to program.config.ts
- rename votingMechanism -> votingStrategy

### Refers/Fixes

closes #154 
closes #155 
closes #156 

### Testing

**Program**

- Deploying Upgradable ProgramFactory to [0xb19589C32351EC32652BAb386b61443b741B678a](https://kovan-optimistic.etherscan.io/address/0xb19589C32351EC32652BAb386b61443b741B678a)
- Deploying ProgramImplementation to [0x6038fd0D126CA1D0b2eA8897a06575100f7b16C2](https://kovan-optimistic.etherscan.io/address/0x6038fd0D126CA1D0b2eA8897a06575100f7b16C2)
- Program created:  [0x857e76d3E83f087c7BA18d03A8d7488fbE52DA1B](https://kovan-optimistic.etherscan.io/address/0x857e76d3E83f087c7BA18d03A8d7488fbE52DA1B)

image

image

====

**Round**

- Deploying Upgradable RoundFactory to [0xbB8f276FE1D52a38FbED8845bCefb9A23138Af92](https://kovan-optimistic.etherscan.io/address/0xbB8f276FE1D52a38FbED8845bCefb9A23138Af92)
- Deploying RoundImplementation to [0xc76Ea06e2BC6476178e40E2B40bf5C6Bf3c40EF6](https://kovan-optimistic.etherscan.io/address/0xc76Ea06e2BC6476178e40E2B40bf5C6Bf3c40EF6)
 - Round created:  [0x3Cd6edA7fDF9ab6b6AF6E226Ce184569C5DF8Ae5](https://kovan-optimistic.etherscan.io/address/0x3Cd6edA7fDF9ab6b6AF6E226Ce184569C5DF8Ae5)

Screenshot 2022-07-19 at 2 58 51 AM

image

### CODE REVIEW

File: packages/contracts/contracts/dummy/upgradeCheck/README.md [hunk 1]
```diff
@@ -26,25 +26,25 @@ User ---- tx ---> Proxy ----------> Implementation_v0
 - Update deploy script.
 
 - Ensure V0 factory contract is deployed.
-    - Command: `yarn run goerli-deploy-program-factory`
+    - Command: `yarn run deploy-program-factory goerli`
```
Comment (QEDK) [inline packages/contracts/contracts/dummy/upgradeCheck/README.md:29]: Change `goerli` to a placeholder like `<network>`?

File: packages/contracts/.vscode/launch.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
             "request": "launch",
             "name": "script: deployRoundImplementation",
             "runtimeExecutable": "yarn",
-            "args": ["goerli-deploy-round"],
```
Comment (QEDK) [inline packages/contracts/.vscode/launch.json:28]: This is gonna be surprising but I do not use VSCode so hard for me to determine what these do. 😆

File: packages/contracts/package.json [hunk 3]
```diff
@@ -12,22 +12,25 @@
     "coverage": "yarn hardhat coverage",
     "export-abi": "yarn run hardhat export-abi",
     "clear-abi": "yarn run hardhat clear-abi",
+
     "// progam-deploy-scripts": "//-- Program deploy Script --//",
```
Comment (QEDK) [inline packages/contracts/package.json:16]: do we gain much by keeping comments as technically valid scripts? might be worthwhile removing altogether or using something like `"// progam-deploy-scripts": "echo \"this script is just a json comment\" 1>&2; exit 1",`

**Review by QEDK — CHANGES_REQUESTED** [review]

File: packages/contracts/contracts/dummy/upgradeCheck/README.md (hunk 1 above)
Comment (thelostone-mc) [reply packages/contracts/contracts/dummy/upgradeCheck/README.md:29]: ^ Oh that's purely an example where we are attempting to deploy goerli.
The actual readme -> state `<network>`

File: packages/contracts/.vscode/launch.json (hunk 2 above)
Comment (thelostone-mc) [reply packages/contracts/.vscode/launch.json:28]: Haha i came across on some random twitter thread. 
This PR should explain what it does but in short -> let's us put in debug points. What editor do you atm ?
https://github.com/gitcoinco/grants-round/pull/23

File: packages/contracts/package.json (hunk 3 above)
Comment (thelostone-mc) [reply packages/contracts/package.json:16]: Oh i like that ! Let me update that.
I just kept it to have a level of grouping (legit doesn't add in value beyond readability) 
I'll update it to your format

File: packages/contracts/.vscode/launch.json (hunk 2 above)
Comment (QEDK) [reply packages/contracts/.vscode/launch.json:28]: Sublime Text 😎

**Review by QEDK — APPROVED** [review]

**Pull request merged**